### PR TITLE
docs: add README for runtime-rs hypervisor crate

### PR DESCRIPTION
**Pull request by Tim-0731-Hzt**
add README for runtime-rs hypervisor crate

Fixes:#4634
Signed-off-by: Zhongtao Hu <zhongtaohu.tim@linux.alibaba.com>

### CODE REVIEW

File: src/runtime-rs/crates/hypervisor/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# Multi-vmm support for runtime-rs
+Some key points for supporting multi-vmm in rust runtime.
+## 1. Hypervisor Config
```
Comment (liubin) [inline src/runtime-rs/crates/hypervisor/README.md:3]: Add a blank line between paragraphs.

File: src/runtime-rs/crates/hypervisor/README.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+The diagram below gives an overvow for hypervisor config
+
+![hypervisor config](../../docs/images/hypervisor-config.svg)
```
Comment (liubin) [inline src/runtime-rs/crates/hypervisor/README.md:6]: Images can be saved in `src/runtime-rs/docs/images/`, `runtime-rs` its own docs directory.

File: src/runtime-rs/crates/hypervisor/README.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+VMM's config info will be loaded when initialize the runtime instance, there are some important functions need to be focused on. 
+### VirtContainer::init()
+[https://github.com/kata-containers/kata-containers/blob/main/src/runtime-rs/crates/runtimes/virt_container/src/lib.rs#L36](https://github.com/kata-containers/kata-containers/blob/main/src/runtime-rs/crates/runtimes/virt_container/src/lib.rs#L36)
```
Comment (liubin) [inline src/runtime-rs/crates/hypervisor/README.md:10]: URL should not include `main`, please use a tag or commit id instead.

File: src/runtime-rs/crates/hypervisor/README.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    async fn check(&self) -> Result<()>;
+    async fn get_jailer_root(&self) -> Result<String>;
+    async fn save_state(&self) -> Result<HypervisorState>;
+   }
+```
+In current design, VM will be started in the following steps.
+
+![vmm start](../../docs/images/vm-start.svg)
+
+## 3. Directory Tree
```
Comment (liubin) [inline src/runtime-rs/crates/hypervisor/README.md:57]: The directory tree is not very useful, and it's easy to be out-of-date. So is it better to delete it?

File: src/runtime-rs/crates/hypervisor/README.md (hunk 2 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/hypervisor/README.md:7]: fixed

File: src/runtime-rs/crates/hypervisor/README.md (hunk 3 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/hypervisor/README.md]: I add code blocks to substitute that

File: src/runtime-rs/crates/hypervisor/README.md (hunk 4 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/hypervisor/README.md]: fixed

File: src/runtime-rs/crates/hypervisor/README.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
+# Multi-vmm support for runtime-rs
+Some key points for supporting multi-vmm in rust runtime.
+## 1. Hypervisor Config
+
+The diagram below gives an overvow for the hypervisor config
```
Comment (quanweiZhou) [inline src/runtime-rs/crates/hypervisor/README.md:5]: typo `overvow`, maybe change to
```
The diagram below gives an overview of the hypervisor config.
```

Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/hypervisor/README.md]: fixed

File: src/runtime-rs/crates/hypervisor/README.md (hunk 2 above)
Comment (liubin) [reply src/runtime-rs/crates/hypervisor/README.md:7]: Oh, I missed the path..

**Review by liubin — APPROVED** [review]

**Review by studychao — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @Tim-0731-Hzt - this is very helpful for #5242! :smile:

lgtm

**Review by quanweiZhou — APPROVED** [review]
Thanks, @Tim-0731-Hzt LGTM

**Pull request merged**